src/App.tsx

```diff
@@ -16,47 +22,84 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [activeButton, setActiveButton] = useState<SortType | null>(null);
+  const [resetVisible, setResetVisible] = useState(false);
+
+  const handleSortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
+
+    setGoods(sortedGoods);
+    setActiveButton(SortType.Alphabetical);
+    setResetVisible(true);
+  };
+
+  const handleSortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(sortedGoods);
+    setActiveButton(SortType.Length);
+    setResetVisible(true);
+  };
+
+  const handleReverse = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setActiveButton(SortType.Reverse);
+    setResetVisible(true);
+  };
+
+  const handleReset = () => {
+    setGoods([...goodsFromServer]);
+    setActiveButton(null);
+    setResetVisible(false);
+  };
```

Comment:
```suggestion
  const handleReset = () => {
    setSortType(false);
    setIsReversed(false);
  };
```

Reply:
sorry, I changed the code before seeing your changes

Reply:
Ok, your current variant looks much better